projects/npm-tools/packages/npm-scripts/src/utils/createFederationConfig.js

```diff
@@ -256,3 +262,15 @@ module.exports = async function () {
 		},
 	};
 };
+
+function prefixExposesPaths(exposes, prefix) {
```

Comment:
A more generic name might be appropriate here, like `transformExposesPaths` or `normalizeExposesPaths`, because `prefix` makes it sound like it's simply going to _add_ to the front of each path, but what it actually does is more more complicated than that (ie. it returns an object, it strips the `.js` extension from the keys, strips the leading `./` from the keys if it is present, and then create a value out of `prefix` and the `filePath`).

Should we throw an error if user passes in a `./` path? I'm thinking of the kind of confusion we talked about [in Slack](https://liferay.slack.com/archives/G3JBR21HA/p1611569317122800) where users can write their `resources/`-relative paths as either `a/b/c` or `./a/b/c`... (forcing them to write it always the same way might be a good thing).

Reply:
I didn't like the name either, but couldn't think of anything better (because of _webpack fatigue_ :sweat_smile: ) so I will use your `normalizeExposesPaths` proposal.

Regarding throwing an error I don't like it too much, because it looks like an anti-feature. I mean, in a external environment (different from`liferay-portal`) it sounds quite reasonable to support both syntaxes (think of a customers that _loves prefixing everything with dots_ using `npm-scripts`). 

So I don't know. Maybe we could move it to the linting layer?

Reply:
I don't think I'd move it to the linting layer. Either we decide it is "wrong" (in which case we should reject it outright):

![reject](https://user-images.githubusercontent.com/7074/105743064-9062c280-5f3c-11eb-9adc-d0b67c75eb83.gif)

Or we decide we don't care (in which case we allow it):

![i-ll-allow-it](https://user-images.githubusercontent.com/7074/105742956-6ad5b900-5f3c-11eb-80e3-5541a3a8ecb5.gif)

(Note that we treat almost all our lint problems as hard errors rather than warnings, because mere warnings/recommendations tend to get ignored and accumulate over time.)

Reply:
Finally renamed it `transform...` :sweat_smile: .

Regarding the check for `./` I will make it mandatory to use `<inputDir>` before the path to make it more explicit. The reason is that, given that we are composing the path from the `build.input`+ `the path the user is configuring`, having a leading `./`is very strange, because it looks like you are referring to the project folder, not `build.input`.

As decided in Slack, we will slowly shift to this paradigm in future PRs so that all the configuration is easily understood by any untrained user.